server/src/main/java/org/opensearch/search/backpressure/trackers/NodeDuressTrackers.java

```diff
@@ -22,24 +24,35 @@
  */
 public class NodeDuressTrackers {
     private final Map<ResourceType, NodeDuressTracker> duressTrackers;
+    private final Map<ResourceType, Boolean> resourceDuressCache = new ConcurrentHashMap<>();
+    private final BooleanSupplier nodeDuressCacheExpiryChecker;
 
     public NodeDuressTrackers(Map<ResourceType, NodeDuressTracker> duressTrackers) {
+        this(duressTrackers, new TimeBasedExpiryTracker(System::nanoTime));
+    }
+
+    public NodeDuressTrackers(Map<ResourceType, NodeDuressTracker> duressTrackers, BooleanSupplier nodeDuressCacheExpiryChecker) {
         this.duressTrackers = duressTrackers;
+        for (ResourceType resourceType : ResourceType.values()) {
+            resourceDuressCache.put(resourceType, false);
+        }
+        this.nodeDuressCacheExpiryChecker = nodeDuressCacheExpiryChecker;
     }
 
     /**
      * Method to check the {@link ResourceType} in duress
      * @return Boolean
      */
     public boolean isResourceInDuress(ResourceType resourceType) {
-        return duressTrackers.get(resourceType).test();
+        return resourceDuressCache.get(resourceType);
```

Comment:
Can we make this cacheable change as an opt-in rather than enforcing for everyone using this tracker?
As search backpressure runs every 1sec by default, so using a cache for this use-case doesn't make much sense. 

Also what if someone plans to only call `isResourceInDuress()` for a specific resource type rather than calling `isNodeInDuress`? Cache would always return false in that case. This can probably be solved by moving updateCache() inside `isResourceInDuress`